### PR TITLE
feat: add pure python implementation

### PR DESCRIPTION
**Pull request by crwilcox**
Default to using CFFI, but if import fails, use a pure python version and warn.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[metadata]
+name = google-crc32c
+version = 0.0.2
+description = A python wrapper of the C library 'Google CRC32C'
```
Comment (busunkim96) [inline setup.cfg]: Is the formatting on this. line intentional?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+url = https://github.com/googleapis/python-crc32c
+long_description = file: README.md
+long_description_content_type = text/markdown
+author = Google LLC
+author_email = googleapis-packages@oogle.com
+
+license = Apache 2.0
+platforms = Posix, MacOS X, Windows
+classifiers =
+    Development Status :: 2 - Pre-Alpha
```
Comment (busunkim96) [inline setup.cfg]: Just double-checking that this is still the correct classifier

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Intended Audience :: Developers
+    License :: OSI Approved :: Apache Software License
+    Operating System :: OS Independent
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.5
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
```
Comment (busunkim96) [inline setup.cfg]: Can a `python_requires` also be added?

File: src/crc32c/_checksum.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    def consume(self, stream, chunksize):
+        """Consume chunks from a stream, extending our CRC32 checksum.
+
+        Args:
+            stream (BinaryIO): the stream to consume.
+            chunksize (int): the size of the read to perform
+
+        Returns:
+            Generator[bytes, None, None]: Tterable of the chunks read from the
```
Comment (busunkim96) [inline src/crc32c/_checksum.py]:
```suggestion
            Generator[bytes, None, None]: Iterable of the chunks read from the
```

File: setup.cfg (hunk 2 above)
Comment (crwilcox) [reply setup.cfg]: This is a good point. I can move it to beta at this point.

File: setup.cfg (hunk 1 above)
Comment (crwilcox) [reply setup.cfg]: I am not sure I understand what you are referring to?
`description = A python wrapper of the C library 'Google CRC32C'`

Comment (busunkim96) [reply setup.cfg]: I was confused by the extra quotes around `Google CRC32C`, but I see how it works now. 😄

**Pull request merged**